core/src/mindustry/entities/comp/UnitComp.java

```diff
@@ -337,6 +338,14 @@ public void update(){
             }
         }
 
+        //apply knockback near rally points
+        for(Tile rally : indexer.getAllied(team, BlockFlag.rally)){
+            float relativeSize = (tilesize * rally.block().size) + hitSize/2f + 1f;
+            if(within(rally.worldx(), rally.worldy(), relativeSize)){
+                vel().add(Tmp.v1.set(this).sub(rally.worldx(), rally.worldy()).setLength(0.1f + 1f - dst(rally) / relativeSize).scl(0.45f * Time.delta));
+            }
+        }
+
```

Comment:
potential issues:
- duplicate code with the above (maybe refractor)
- rally.block() might be null sometimes (like constructing)
- rally.worldx() seems to use the corner instead of the center